normandy/recipes/forms.py

```diff
@@ -0,0 +1,38 @@
+import json
+
+from django import forms
+from django.core.exceptions import ValidationError
+
+import jsonschema
+
+
+class ActionAdminForm(forms.ModelForm):
+    def clean(self):
+        """
+        Validate that the action being edited isn't currently in use.
+        """
+        super().clean()
+
+        if self.instance.in_use:
+            self.add_error(
+                None,
+                ValidationError('Action is currently in use and cannot be edited.')
+            )
+
+
+class RecipeActionInlineForm(forms.ModelForm):
+    def clean(self):
+        """Validate the arguments against their schema."""
```

Comment:
I'm a little nervous about making this specific to the admin interface (and other things that use forms). Although right now this is the only way that recipes are created and edited, I can certainly image other ways in the future. Namely I'm thinking of API access to this, which I imagine we will want eventually.

One way we could handle this is to re-implement the schema validation logic in the API. I'm not excited about this. Another option would be to put this validation into the model somehow. That makes nice error messages harder though, I imagine.

Is this something you want to worry about now? Or should we just kick it down the road until we actually have another editor of these models?

Also, tests for validation would be nice.